### PR TITLE
6 invite team to quiz

### PR DESCRIPTION
**Pull request by witterless**

### CODE REVIEW

File: src/test/java/com/gavinfenton/quizolation/controller/QuizControllerTest.java [hunk 1]
```diff
@@ -68,6 +66,12 @@ public void testCreateQuizCallsAndReturnsQuizFromService() throws Exception {
         assertEquals(quizExpected, quizActual);
     }
 
+    @Test
+    public void testAddTeamToQuizCallsAndReturnsFromService() {
+
+
+    }
+
```
Comment (GavinF17) [inline src/test/java/com/gavinfenton/quizolation/controller/QuizControllerTest.java]: Guessing this was to be deleted?

File: src/test/java/com/gavinfenton/quizolation/service/QuizServiceTest.java [hunk 2]
```diff
@@ -144,6 +148,29 @@ public void testUpdateQuizSetsIdBeforeSaveAndReturnsFromRepository() {
         assertEquals(quizExpected, quizActual);
     }
 
+    @Test
+    public void testAddTeamToQuizCallsAndReturnsFromRepo() throws Exception {
+        //Given
+        Quiz quizSaving = new Quiz();
+        Long quizIdSavingExpected = 32L;
```
Comment (GavinF17) [inline src/test/java/com/gavinfenton/quizolation/service/QuizServiceTest.java]: Think this name is confusing, quizIdSaving would be clearer, also applies to teamIdSavingExpected

File: src/test/java/com/gavinfenton/quizolation/service/QuizServiceTest.java [hunk 3]
```diff
@@ -144,6 +148,29 @@ public void testUpdateQuizSetsIdBeforeSaveAndReturnsFromRepository() {
...
+        Long teamIdSavingExpected = 1L;
+        Team teamExpected = new Team();
+        given(teamService.getTeam(teamIdSavingExpected)).willReturn(teamExpected);
+        given(quizRepository.findById(quizIdSavingExpected)).willReturn(Optional.of(quizSaving));
+        given(quizRepository.save(quizSaving)).willReturn(quizExpected);
+
+        //When
+        Quiz quizActual = quizService.addTeamToQuiz(quizIdSavingExpected, teamIdSavingExpected);
+
+        //Then
```
Comment (GavinF17) [inline src/test/java/com/gavinfenton/quizolation/service/QuizServiceTest.java:168]: I would add a check to make sure that team has been added to the Quiz returned from getQuiz()

File: src/test/java/com/gavinfenton/quizolation/service/QuizServiceTest.java [hunk 4]
```diff
@@ -144,6 +148,29 @@ public void testUpdateQuizSetsIdBeforeSaveAndReturnsFromRepository() {
         assertEquals(quizExpected, quizActual);
     }
 
+    @Test
+    public void testAddTeamToQuizCallsAndReturnsFromRepo() throws Exception {
+        //Given
+        Quiz quizSaving = new Quiz();
+        Long quizIdSavingExpected = 32L;
+        Quiz quizExpected = new Quiz();
+        Team teamSaving = new Team();
```
Comment (GavinF17) [inline src/test/java/com/gavinfenton/quizolation/service/QuizServiceTest.java]: This is unused

File: src/test/java/com/gavinfenton/quizolation/service/QuizServiceTest.java [hunk 5]
```diff
@@ -144,6 +148,29 @@ public void testUpdateQuizSetsIdBeforeSaveAndReturnsFromRepository() {
...
+        given(quizRepository.save(quizSaving)).willReturn(quizExpected);
+
+        //When
+        Quiz quizActual = quizService.addTeamToQuiz(quizIdSavingExpected, teamIdSavingExpected);
+
+        //Then
+        verify(teamService).getTeam(teamIdSavingExpected);
+        verify(quizRepository).findById(quizIdSavingExpected);
+        verify(quizRepository).save(quizSaving);
+        assertEquals(quizExpected, quizActual);
```
Comment (GavinF17) [inline src/test/java/com/gavinfenton/quizolation/service/QuizServiceTest.java:172]: quizSaving and quizExpected could have the same hashcode value as no differences have been guaranteed by updating a field on one of them in Given

**Review by GavinF17 — CHANGES_REQUESTED** [review]

File: src/test/java/com/gavinfenton/quizolation/controller/QuizControllerTest.java (hunk 1 above)
Comment (witterless) [reply src/test/java/com/gavinfenton/quizolation/controller/QuizControllerTest.java]: possibly yes

File: src/test/java/com/gavinfenton/quizolation/service/QuizServiceTest.java (hunk 2 above)
Comment (witterless) [reply src/test/java/com/gavinfenton/quizolation/service/QuizServiceTest.java]: Resolved

File: src/test/java/com/gavinfenton/quizolation/service/QuizServiceTest.java (hunk 4 above)
Comment (witterless) [reply src/test/java/com/gavinfenton/quizolation/service/QuizServiceTest.java]: Removed

File: src/test/java/com/gavinfenton/quizolation/service/QuizServiceTest.java (hunk 5 above)
Comment (witterless) [reply src/test/java/com/gavinfenton/quizolation/service/QuizServiceTest.java:172]: Resolved (i think)

File: src/test/java/com/gavinfenton/quizolation/service/QuizServiceTest.java (hunk 3 above)
Comment (witterless) [reply src/test/java/com/gavinfenton/quizolation/service/QuizServiceTest.java:168]: resolved

File: src/test/java/com/gavinfenton/quizolation/service/QuizServiceTest.java (hunk 5 above)
Comment (GavinF17) [reply src/test/java/com/gavinfenton/quizolation/service/QuizServiceTest.java:172]: Nope, quizExpected should have something set on it to make sure that an empty Quiz isn't returned

File: src/test/java/com/gavinfenton/quizolation/service/QuizServiceTest.java [hunk 6]
```diff
@@ -144,6 +148,32 @@ public void testUpdateQuizSetsIdBeforeSaveAndReturnsFromRepository() {
         assertEquals(quizExpected, quizActual);
     }
 
+    @Test
+    public void testAddTeamToQuizCallsAndReturnsFromRepo() throws Exception {
```
Comment (GavinF17) [inline src/test/java/com/gavinfenton/quizolation/service/QuizServiceTest.java]: Just spotted, throws Exception isn't necessary as no exception will be thrown

Comment (witterless) [reply src/test/java/com/gavinfenton/quizolation/service/QuizServiceTest.java]: Done

**Review by GavinF17 — APPROVED** [review]

**Pull request merged**